File: SQL/Expressions/SQLRightFunctionExpression.vb
﻿' ___________________________________________________
'
'  (c) Hi-Integrity Systems 2012. All rights reserved.
'  www.hisystems.com.au<http://www.hisystems.com.au> - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On

Namespace SQL

    Public Class SQLRightFunctionExpression
        Inherits SQLFunctionExpression

        Public Sub New(ByVal strFieldName As String, ByVal intLength As Integer)

            Me.New(New SQLFieldExpression(strFieldName), intLength)

        End Sub

        Public Sub New(ByVal objExpression As SQLExpression, ByVal intLength As Integer)

            MyBase.New("RIGHT", objExpression, New SQLValueExpression(intLength))

            If intLength < 0 Then
                Throw New ArgumentException("Length: " & intLength)
            End If

        End Sub

    End Class

End Namespace

File: SQL/Expressions/SQLRightFunctionExpression.cs
// ___________________________________________________
//
//  (c) Hi-Integrity Systems 2012. All rights reserved.
//  www.hisystems.com.au<http://www.hisystems.com.au> - Toby Wicks
// ___________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.SQL
{
	public class SQLRightFunctionExpression : SQLFunctionExpression
	{
		public SQLRightFunctionExpression(string strFieldName, int intLength) 
            : this(new SQLFieldExpression(strFieldName), intLength)
		{
		}
			
		public SQLRightFunctionExpression(SQLExpression objExpression, int intLength) 
            : base(objExpression, new SQLValueExpression(intLength))
		{
			if (intLength < 0)
				throw new ArgumentException("Length: " + intLength.ToString());
		}

		internal override string SQL(Serializers.Serializer serializer)
		{
			return serializer.SerializeRightFunctionExpression(this);
		}
	}
}
